vllm/v1/engine/core.py

```diff
@@ -681,17 +681,19 @@ def _perform_handshake(
             # external LB case for our colocated front-end to use (coordinator
             # only runs with rank 0).
             dp_stats_address = self.frontend_stats_publish_address
-            handshake_socket.send(
-                msgspec.msgpack.encode(
-                    {
-                        "status": "READY",
-                        "local": local_client,
-                        "headless": headless,
-                        "num_gpu_blocks": num_gpu_blocks,
-                        "dp_stats_address": dp_stats_address,
-                    }
-                )
-            )
+
+            # Include config hash for DP configuration validation
+            ready_msg = {
+                "status": "READY",
+                "local": local_client,
+                "headless": headless,
+                "num_gpu_blocks": num_gpu_blocks,
+                "dp_stats_address": dp_stats_address,
+            }
+            if vllm_config.parallel_config.data_parallel_size > 1:
+                ready_msg["config_hash"] = vllm_config.parallel_config.compute_hash()
```

Comment:
Suggest to rename this for clarity
```suggestion
                ready_msg["parallel_config_hash"] = vllm_config.parallel_config.compute_hash()
```